.github/workflows/tests.yaml

```diff
@@ -8,23 +8,57 @@ on:
     - master
 jobs:
   test:
-    runs-on: ubuntu-latest
+    name: ${{ matrix.name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+
+          - name: latest supported versions
+            os: ubuntu-latest
+            python: 3.8
+            toxenv: py38-test-all-latest-cov
+            toxposargs: --cov-report xml
+            coverage: true
+
+          - name: oldest supported versions
+            os: ubuntu-latest
+            python: 3.6
+            toxenv: py36-test-oldest
+
+          - name: latest experimental versions
+            os: ubuntu-latest
+            python: 3.8
+            toxenv: py38-test
```

Comment:
```suggestion
            toxenv: py38-test-dev
```

Reply:
I have tried that, but it needs dependencies for building the modules. I don't think we need to support the dev versions necessarily, and I think it's ok if, most of the time, 'latest supported' and 'latest experimental' are the same versions.

Reply:
Numpy and scipy build 'nightly' test wheels: https://anaconda.org/scipy-wheels-nightly which is what astropy use in their `devdeps` testing environment

Reply:
I'd much rather try and get that to work so there is a meaningful difference between `latest` and `dev`

Reply:
I agree on getting `dev` to work, I'm not convinced about testing against unstable nightlies. I think the latest unsupported tests are helpful when new versions of numpy, scipy, astropy come out and we need to see where we stand, as happened now with scipy 1.5.